kubetest2-gke/deployer/options/project.go

```diff
@@ -19,9 +19,9 @@ package options
 type ProjectOptions struct {
 	Projects []string `flag:"~project" desc:"Comma separated list of GCP Project(s) to use for creating the cluster."`
 
-	BoskosLocation                 string `flag:"~boskos-location" desc:"If set, manually specifies the location of the Boskos server."`
-	BoskosResourceType             string `flag:"~boskos-resource-type" desc:"If set, manually specifies the resource type of GCP projects to acquire from Boskos."`
-	BoskosAcquireTimeoutSeconds    int    `flag:"~boskos-acquire-timeout-seconds" desc:"How long (in seconds) to hang on a request to Boskos to acquire a resource before erroring."`
-	BoskosHeartbeatIntervalSeconds int    `flag:"~boskos-heartbeat-interval-seconds" desc:"How often (in seconds) to send a heartbeat to Boskos to hold the acquired resource. 0 means no heartbeat."`
-	BoskosProjectsRequested        int    `flag:"~projects-requested" desc:"Number of projects to request from Boskos. It is only respected if projects is empty, and must be larger than zero."`
+	BoskosLocation                 string   `flag:"~boskos-location" desc:"If set, manually specifies the location of the Boskos server."`
+	BoskosAcquireTimeoutSeconds    int      `flag:"~boskos-acquire-timeout-seconds" desc:"How long (in seconds) to hang on a request to Boskos to acquire a resource before erroring."`
+	BoskosHeartbeatIntervalSeconds int      `flag:"~boskos-heartbeat-interval-seconds" desc:"How often (in seconds) to send a heartbeat to Boskos to hold the acquired resource. 0 means no heartbeat."`
+	BoskosResourceType             []string `flag:"~boskos-resource-type" desc:"If set, manually specifies the resource type(s) of GCP projects to acquire from Boskos."`
```

Comment:
what is the purpose of this change?

Reply:
For multiproject multicluster topology with shared VPC network, we need at least 3 projects - one is the host project and the other two are the service projects. A host project cannot be used as a service project, and vice versa, so we have to acquire them from 2 different project pools, one from the host project pool and the other two from the service project pool.

Currently for ASM testing we are acquiring these projects out of the kubetest2 gke deployer, and this change is to enable us to acquire the projects with kubetest2.